### PR TITLE
👷🏻 Add CI to build, test and tidy

### PR DESCRIPTION
**Pull request by francoisdtm**

### CODE REVIEW

**Review by e-n-0 — DISMISSED** [review]

File: cmd/dc-supervisor/root.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func init() {
 }
 
 // Starts cobra
-func Execute() {
+func main() {
 	if err := rootCmd.Execute(); err != nil {
 		fmt.Fprint(os.Stderr, err)
```
Comment (e-n-0) [inline cmd/dc-supervisor/root.go:39]: May prefer the use of `FatalLn()` for command errors.

```suggestion
		fmt.FatalLn(os.Stderr, err)
```

Comment (francoisdtm) [reply cmd/dc-supervisor/root.go:39]: `log.Fatalln` equals `Println()` followed by `os.Exit(1)`.
We should either use `log.Fatalln` everywhere and remove`os.Exit(1)`, or keep it as is.

Comment (e-n-0) [reply cmd/dc-supervisor/root.go:39]: or using `Errorln()`

```suggestion
		log.Errorln(err)
```

Comment (francoisdtm) [reply cmd/dc-supervisor/root.go:39]: Using `logrus` to print CLI related error is not wanted.

**Review by e-n-0 — APPROVED** [review]

**Pull request merged**